lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -585,15 +584,22 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
         return;
 
       case 'flutter/web_test_e2e':
-        const MethodCodec codec = JSONMethodCodec();
         replyToPlatformMessage(
             callback,
-            codec.encodeSuccessEnvelope(
-                _handleWebTestEnd2EndMessage(codec, data)));
+            jsonCodec.encodeSuccessEnvelope(
+                _handleWebTestEnd2EndMessage(jsonCodec, data)));
         return;
 
       case 'flutter/platform_views':
-        implicitView!.platformViewMessageHandler.handlePlatformViewCall(data, callback!);
+        final MethodCall(:String method, :dynamic arguments) = standardCodec.decodeMethodCall(data);
+        if (arguments is Map<dynamic, dynamic>) {
+          final int? flutterViewId = arguments.tryInt('flutterViewId');
+          if (flutterViewId != null) {
+            viewData[flutterViewId]!.platformViewMessageHandler.handlePlatformViewCall(method, arguments, callback!);
+            return;
+          }
+        }
```

Comment:
The bit that extracts `flutterViewId` from `arguments` should probably be a standard method that we reuse (similar to the codec), so we end up with something like:

```dart
final MethodCall ... arguments = xxCodec.decodeMethodCall(data);
final int? flutterViewId = extractViewId(arguments);
if (flutterViewId != null) {
  // new stuff
} else {
  // legacy stuff
} 
```

I think that there are more platform dispatcher methods that are going to pass a `flutterViewId` parameter, like: `render` and `updateSemantics` (at least), so we could reuse the same method for all.

Reply:
Done!